Cython/Utility/ObjectHandling.c

```diff
@@ -2214,12 +2214,52 @@ static PyObject* __Pyx_PyObject_CallMethod1(PyObject* obj, PyObject* method_name
 
 
 /////////////// tp_new.proto ///////////////
+//@requires: ModuleSetupCode.c::GetRuntimeVersion
 
 #define __Pyx_tp_new(type_obj, args) __Pyx_tp_new_kwargs(type_obj, args, NULL)
+#if CYTHON_COMPILING_IN_LIMITED_API && __PYX_LIMITED_VERSION_HEX < 0x030A0000
+static CYTHON_INLINE PyObject* __Pyx_tp_new_fallback(PyObject* type_obj, PyObject* args, PyObject* kwargs); /*proto*/
```

Comment:
```suggestion
static PyObject* __Pyx_tp_new_fallback(PyObject* type_obj, PyObject* args, PyObject* kwargs); /*proto*/
```

Not inline.

Reply:
Why other function (with split proto and impl) in the exact same file use CYTHON_INLINE then? Look at PyObjectCall, for example? What's the criteria for using or not using CYTHON_INLINE?

Reply:
Either will work so it's not a hard suggestion. But if we've moved it out of the proto section specifically because we think it's too bulky to inline then it seems a bit silly to ask the compiler to inline it.

Reply:
As my fallback is used in Py_LIMITED_API, which is already slower that normal mode, I'm not convinced that the proto/impl split requested by @scoder is really that crucial. My last changes with the split had the sole intention to humor him, now I'll remove CYTHON_INLINE to humor you.

PS: A really large number of functions in `ObjectHandling.c` use CYTHON_INLINE. Some of these functions look more complicated that the silly fallback I implemented. What's the criteria to use or not use CYTHON_INLINE in utility code?

Reply:
> What's the criteria to use or not use CYTHON_INLINE in utility code?

There's more than one. In some cases, we use(d) `CYTHON_INLINE` on functions for which it's difficult to decide whether they will be used or not. As long as they compile down to something simple, the inlining request (which is not a strict command, more of a hint to the C compiler) won't hurt but the C compiler also won't complain that they are unused.

More importantly, I broadly used `CYTHON_INLINE` on functions that receive many constant arguments, so that inlining them can strip down their effective binary code considerably. In some extreme cases, a relatively long function can be boiled down to just a few instructions, depending on the usage. And there are several examples where I split up the overall code into multiple functions, only the hottest ones of which are marked as `CYTHON_INLINE`, to lay out the fast path for the C compiler.

In this PR, we know when (and that) the function is used and that the fallback function is absolutely not worth inlining, because it's relatively expensive and almost never used in modern CPython runtimes. So, I fail to see any reason to request inlining.